### PR TITLE
Fitness tracker#214

### PR DESCRIPTION
**Pull request by CrypticRevenger**
# Title and Issue number 
Add Fitness Tracker #214 
Title : Add Fitness Tracker
Issue No. : #214

# Checklist:

- [x] I have mentioned the issue number in my Pull Request.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have updated the Index.html file for my contribution

### CODE REVIEW

**Comment by CrypticRevenger** [comment]
@sadanandpai sorry for the new PR, I did the same that you told but, a new PR is created. I apologize.
Thank you.

File: native/helpers/challenges.js [hunk 1]
```diff
@@ -453,4 +453,12 @@ export const challenges = [
   { title: 'Emoji Editor', link: '#' },
   { title: 'Sort Table', link: '#' },
   { title: 'Infinite Scroll', link: '#' },
+  {
+    title: 'Fitness Tracker',
+    link: 'fitness-tracker',
```
Comment (arpansaha13) [inline native/helpers/challenges.js:458]: This link is incorrect in the UI
Either make the link `fitness-tracker/tracker.html`, or rename the `tracker.html` file to `index.html`

Comment (arpansaha13) [reply native/helpers/challenges.js:458]: Also place the challenges with same difficulty together in the challenges list

**Comment by CrypticRevenger** [comment]
@sadanandpai , @arpansaha13  All the changes are done as per your instructions.
Thank you.

**Review by arpansaha13 — COMMENTED** [review]
I think there is a problem probably with the CSS. Please open this file in your local and fix the navbar. Navbar should be on top like in other challenges.

File: native/machine-coding/fitness-tracker/style.css [hunk 2]
```diff
@@ -0,0 +1,135 @@
+body {
+    background-color: #f5f5f5;
+    padding:10vh;
+    display: flex;
+    justify-content: center;
+    align-items: center;
```
Comment (arpansaha13) [inline native/machine-coding/fitness-tracker/style.css:6]: The navbar won't appear on screen properly due to flexbox in the body.

File: native/machine-coding/fitness-tracker/index.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        <button id="dark-mode-toggle" onclick="toggleDarkMode()">Mode</button>
+
+        <!-- Fitness Tracker Content -->
+        <div id="workout-form">
+            <h2>Add a Workout</h2>
+            <label for="workout">Workout Type: </label>
+            <input type="text" id="workout">
+            <label for="duration">Duration (minutes): </label>
+            <input type="number" id="duration">
+            <button onclick="addWorkout()">Add</button>
```
Comment (arpansaha13) [inline native/machine-coding/fitness-tracker/index.html:21]: Looks like none of the js functions are working. You may try using `addEventListener` in the script instead.

**Comment by CrypticRevenger** [comment]
@arpansaha13 Fixed all the above problems.
Thank you.

**Comment by arpansaha13** [comment]
You should remove onclick handlers from html
All event listeners should be added from the script using addEventListener

**Comment by CrypticRevenger** [comment]
@sadanandpai @arpansaha13 All the problems are fixed.
Thank you.

File: native/machine-coding/fitness-tracker/index.html [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+        </div>
+        <div id="workout-list">
+            <h2>Workout Log</h2>
+            <ul id="workouts">
+                <!-- Workout items will be displayed here -->
+            </ul>
+        </div>
+        <p id="error-message" style="color: #e74c3c;"></p> <!-- Add the error message element -->
+    </div>
+    <script src="index.js"></script>
```
Comment (sadanandpai) [inline native/machine-coding/fitness-tracker/index.html:30]: move to head with defer attribute

File: native/machine-coding/fitness-tracker/index.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
+// JavaScript code for the fitness tracker
+document.addEventListener("DOMContentLoaded", function () {
```
Comment (sadanandpai) [inline native/machine-coding/fitness-tracker/index.js:2]: wrapping in DOMContentLoaded listener is not needed when script is loaded with defer attirbute

**Review by sadanandpai — CHANGES_REQUESTED** [review]

**Comment by CrypticRevenger** [comment]
@sadanandpai All the issues are fixed.
Thank you.

**Pull request merged**